### PR TITLE
[ELY-693] SPNEGO continuation required fixes

### PR DESCRIPTION
**Pull request by honza889**
SPNEGO authentication reworked to fulfil test of mchoma.
https://issues.jboss.org/browse/ELY-693

### CODE REVIEW

File: src/main/java/org/wildfly/security/http/impl/SpnegoAuthenticationMechanism.java [hunk 1]
```diff
@@ -99,71 +100,78 @@ public void evaluateRequest(HttpServerRequest request) throws HttpAuthentication
...
-            List<String> authorizationValues = request.getRequestHeaderValues(AUTHORIZATION);
-            Optional<String> challenge = authorizationValues != null ? authorizationValues.stream()
-                    .filter(s -> s.startsWith(CHALLENGE_PREFIX)).limit(1).map(s -> s.substring(CHALLENGE_PREFIX.length()))
-                    .findFirst() : Optional.empty();
-
-            if (log.isTraceEnabled()) {
-                log.tracef("Sent HTTP authorizations: [%s]", authorizationValues == null ? "null" : String.join(", ", authorizationValues));
-            }
+            try {
+                gssContext = GSSManager.getInstance().createContext(gssCredential);
```
Comment (dmlloyd) [inline src/main/java/org/wildfly/security/http/impl/SpnegoAuthenticationMechanism.java]: Maybe this isn't the right time to notice this, but this mechanism (and its factory) should probably accept a GSSManager parameter whose default value is GSSManager.getInstance().  See org.wildfly.security.sasl.gs2.Gs2SaslServerFactory for reference.

**Comment by dmlloyd** [comment]
Otherwise this change seems OK to me

File: src/main/java/org/wildfly/security/http/impl/SpnegoAuthenticationMechanism.java (hunk 1 above)
Comment (darranl) [inline src/main/java/org/wildfly/security/http/impl/SpnegoAuthenticationMechanism.java]: @honza889 could you please create a Jira for David's suggestion?

Comment (honza889) [inline src/main/java/org/wildfly/security/http/impl/SpnegoAuthenticationMechanism.java]: https://issues.jboss.org/browse/ELY-709

**Pull request merged**